From 3a000b9052e7003de22fee52c07cb75695351f89 Mon Sep 17 00:00:00 2001
From: Enrico Jorns <ejo@pengutronix.de>
Date: Tue, 16 Jan 2018 15:34:41 +0100
Subject: [PATCH 29/44] src/install: skip checksum verification for
 casync-based images

The checksum and size of casync 'images' as located in bundle is only
the checksum/size of the index file, not the size of the image itself.

Signed-off-by: Enrico Jorns <ejo@pengutronix.de>
---
 src/install.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/install.c b/src/install.c
index 83e0097..301cfa2 100644
--- a/src/install.c
+++ b/src/install.c
@@ -848,11 +848,13 @@ static gboolean launch_and_wait_default_handler(RaucInstallArgs *args, gchar* bu
 			goto out;
 		}
 
-		/* Verify image checksum */
-		res = verify_checksum(&mfimage->checksum, mfimage->filename, &ierror);
-		if (!res) {
-			g_propagate_prefixed_error(error, ierror, "Failed verifying checksum: ");
-			goto out;
+		/* Verify image checksum (for non-casync images) */
+		if (!g_str_has_suffix(mfimage->filename, ".caibx") && !g_str_has_suffix(mfimage->filename, ".caidx")) {
+			res = verify_checksum(&mfimage->checksum, mfimage->filename, &ierror);
+			if (!res) {
+				g_propagate_prefixed_error(error, ierror, "Failed verifying checksum: ");
+				goto out;
+			}
 		}
 
 		/* determine whether update image type is compatible with destination slot type */
-- 
2.14.1

